tensorboard/plugins/debugger_v2/tf_debugger_v2_plugin/effects/debugger_effects_test.ts

```diff
@@ -881,6 +907,81 @@ describe('Debugger effects', () => {
     }
   });
 
+  describe('graphExecutionScrollToIndex', () => {
+    beforeEach(() => {
+      debuggerEffects.loadData$.subscribe();
+    });
+
+    for (const {dataExists, page3Size, loadingPages} of [
+      {dataExists: false, page3Size: 0, loadingPages: [3]},
+      {dataExists: false, page3Size: 0, loadingPages: []},
+      {dataExists: true, page3Size: 2, loadingPages: []},
+    ]) {
+      it('triggers GraphExecution loading', fakeAsync(() => {
+        const runId = '__default_debugger_run__';
+        const originalScrollBeginIndex = 50;
+        const newScrollBeginIndex = originalScrollBeginIndex + 2;
+        const numGraphExecutions = 100;
+        const pageSize = 20;
+        const displayCount = 10;
+        store.overrideSelector(getActiveRunId, runId);
+        store.overrideSelector(getNumGraphExecutions, numGraphExecutions);
+        store.overrideSelector(
+          getGraphExecutionScrollBeginIndex,
+          newScrollBeginIndex
+        );
+        store.overrideSelector(getGraphExecutionPageSize, pageSize);
+        store.overrideSelector(getGraphExecutionDisplayCount, displayCount);
+        store.overrideSelector(getExecutionPageSize, pageSize);
+        store.overrideSelector(getGraphExecutionDataLoadingPages, loadingPages);
+        const pageLoadedSizes: {[pageIndex: number]: number} = {
+          0: 20,
+          1: 20,
+          2: 20,
+        };
+        pageLoadedSizes[3] = page3Size;
+        store.overrideSelector(
+          getGraphExecutionDataPageLoadedSizes,
+          pageLoadedSizes
+        );
+        store.refreshState();
+
+        const graphExecutions = new Array<GraphExecution>(pageSize).fill(
+          createTestGraphExecution()
+        );
+        const graphExecutionDataResponse: GraphExecutionDataResponse = {
+          begin: 60,
+          end: 60 + pageSize,
+          graph_executions: graphExecutions,
+        };
+        const fetchExecutionData = createFetchGraphExecutionDataSpy(
+          runId,
+          60,
+          60 + pageSize,
+          graphExecutionDataResponse
+        );
+
+        action.next(graphExecutionScrollToIndex({index: newScrollBeginIndex}));
+        tick(100);
```

Comment:
`tick(100)` is needed because of the `debounceTime(100)` in the effect created by `onGraphExecutionScroll()`. This test won't pass without it. `100` is therefore not an arbitrary number. 

Using `fakeAsync()` with `tick()` is better than `async/await` because it is *simulated* passage of time and hence should make the test run in shorter period of time.